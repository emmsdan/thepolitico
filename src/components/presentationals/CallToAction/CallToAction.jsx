import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const CallToAction = ({ withButton }) => {
  return (
    <section className="call-to-action">
      <div className="placeholder">
        <h2>Every Vote Count</h2>
        <h4>Vote and Be Voted for</h4>
        {withButton && (
          <Link to="login" className="default-button">
            Register Today
          </Link>
        )}
      </div>
    </section>
  );
};

CallToAction.propTypes = {
  withButton: PropTypes.bool,
};

export default CallToAction;
