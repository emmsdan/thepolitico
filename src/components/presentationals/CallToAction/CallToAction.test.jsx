import React from 'react';
import { shallow } from 'enzyme';

import CallToAction from './CallToAction';

describe('<CallToAction />', () => {
  it('should render correctly without button', () => {
    const component = shallow(<CallToAction />);
    expect(component).toMatchSnapshot();
  });

  it('should render correctly with button', () => {
    const component = shallow(<CallToAction withButton={true} />);
    expect(component).toMatchSnapshot();
  });
});
