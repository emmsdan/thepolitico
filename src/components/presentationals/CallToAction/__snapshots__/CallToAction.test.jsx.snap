// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CallToAction /> should render correctly with button 1`] = `
<section
  className="call-to-action"
>
  <div
    className="placeholder"
  >
    <h2>
      Every Vote Count
    </h2>
    <h4>
      Vote and Be Voted for
    </h4>
    <Link
      className="default-button"
      to="login"
    >
      Register Today
    </Link>
  </div>
</section>
`;

exports[`<CallToAction /> should render correctly without button 1`] = `
<section
  className="call-to-action"
>
  <div
    className="placeholder"
  >
    <h2>
      Every Vote Count
    </h2>
    <h4>
      Vote and Be Voted for
    </h4>
  </div>
</section>
`;
